archivist/cmds/usage/run.py

```diff
@@ -0,0 +1,153 @@
+"""Run RKVST usage summary script."""
+
+from contextlib import suppress
+from logging import getLogger
+from sys import exit as sys_exit
+
+from os import path, makedirs, remove, getcwd
+from json import dump as json_dump
+from datetime import datetime
+from shutil import move, make_archive
+
+from archivist.dictmerge import (
+    assets_ext_attr,
+    attachment_identities_assets,
+    events_ext_attr,
+    attachment_identities_events,
+    _deepmerge,
+)
+
+LOGGER = getLogger(__name__)
+
+
+def run(arch, args):
+    """Run diagnostic script."""
+    props = {"confirmation_status": "CONFIRMED"}
+    attrs = {}  # attributes can be added to filer by name, type, etc.
+
+    # Total Number of Assets
+    assets = list(arch.assets.list(props=props, attrs=attrs))
+
+    # Assets with Extended Attributes
+    extended_attributes_asset = assets_ext_attr(assets)
+
+    # Assets with Associated Location
+    assets_with_location = list(
+        arch.assets.list(
+            props=props,
+            attrs=_deepmerge(
+                attrs,
+                [
+                    a
+                    for a in assets
+                    if a.get("attributes", {}).get("arc_home_location_identity")
+                ],
+            ),
+        )
+    )
+
+    # Assets with Attachments
+    assets_with_attachments = list(
+        arch.assets.list(
```

Comment:
Ditto - simply feed assets variable thru a list comprehension here instead of exercising the API again

Reply:
add to dictmerge.py and add unittests

Reply:
Changed to single list comprehension and unit tests added